File: Source/CookItNow.Business/Models/QuickRecipe.cs
using System;
using System.Collections.Generic;

namespace CookItNow.Business.Models
{
    public class QuickRecipe
	{
        public QuickRecipe()
        {
            this.Durations = new List<Duration>();
            this.Subrecipes = new List<Subrecipe>();
            this.Ingredients = new List<Ingredient>();
            this.Steps = new List<Step>();
        }

		public string Id { get; set; }
		
		public string Title { get; set; }

        public string OriginalUrl { get; set; }
		
		public string Summary { get; set; }
		
		public string OriginalServings { get; set; }
		
		public IList<Duration> Durations { get; set; }

        public IList<Subrecipe> Subrecipes { get; set; }

        public IList<Ingredient> Ingredients { get; set; }

        public IList<Step> Steps { get; set; }
	}
}

File: Source/CookItNow.Web/app/quick-recipe/models/quick-recipe.ts
import {Ingredient} from "shared/models/ingredient"; 

export interface QuickRecipe {
    id : number;
    language: string;
    title : string;
    originalUrl : string;
    summary : string;
    originalServings : string;
    duration : Duration[];
    subrecipes : SubRecipe[];
    ingredients : Ingredient[];
    steps: Step[];
}

export interface Duration {
    title: string;
    time: string;
}

export interface SubRecipe {
    id: number;
    title: string;
}

export interface Step {
    subrecipeId: number;
    parts: Part[];
    postStep: Step;
}

export interface Part {
    type: string;
}

export interface IngredientPart extends Part {
    ingredient: Ingredient;
}

export interface TextPart extends Part {
    value: string;
}

export interface ActionPart extends Part {
    value: string;
}

export interface TimerPart extends Part {
    value: string;
    action: string;
}

export interface IngredientEnumerationPart extends Part {
    ingredients: Ingredient[];
}
